### PR TITLE
document axhline from hlines docstring

### PR DESCRIPTION
**Pull request by pierre-haessig**
I propose these small additions in the docstrings ("see also" sections) of the axhlines, axhspan and hlines family of axes methods to increase the discoverability of axhline (and same for axvline of course). Indeed, it's been years I've been working around `ax.hlines` to get lines which spans the entire axis... only to discover today `ax.axhline` out of pure chance, by reading the [Transformations Tutorial](https://matplotlib.org/users/transforms_tutorial.html).

In the end I tried to write a coherent and useful set of cross references between the six methods ( axhlines, axhspan and hlines, and vertical counterparts). I've favored compactness over exhaustivity.

### CODE REVIEW

File: lib/matplotlib/axes/_axes.py [hunk 1]
```diff
@@ -705,7 +705,8 @@ def axhline(self, y=0, xmin=0, xmax=1, **kwargs):
 
         See also
         --------
-        axhspan : for example plot and source code
+        hline : add horizontal lines in data coordinates
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: Uh, this should be `hlines` ...

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -771,7 +772,8 @@ def axvline(self, x=0, ymin=0, ymax=1, **kwargs):
 
         See also
         --------
-        axhspan : for example plot and source code
+        vline : add vertical lines in data coordinates
```
Comment (QuLogic) [inline lib/matplotlib/axes/_axes.py]: ... and this should be `vlines`.

**Comment by pierre-haessig** [comment]
Ok, I've updated my branch [axlines-doc](https://github.com/pierre-haessig/matplotlib/tree/axlines-doc), but since this PR is merged, they don't appear here. Should I make a new PR?

**Comment by WeatherGod** [comment]
yes, please.

**Pull request merged**